developer/js/source/lexical-model-compiler/build-trie.ts

```diff
@@ -380,31 +383,35 @@ namespace Trie {
         .reduce((acc, count) => acc + count, 0);
     }
   }
+}
 
-  /**
-   * Converts word forms in into an indexable form. It does this by converting
-   * the string to uppercase and trying to remove diacritical marks.
-   *
-   * This is a very naïve implementation, that I've only though to work on
-   * languages that use the Latin script. Even then, some Latin-based
-   * orthographies use code points that, under NFD normalization, do NOT
-   * decompose into an ASCII letter and a combining diacritical mark (e.g.,
-   * SENĆOŦEN).
-   *
-   * Use this only in early iterations of the model. For a production lexical
-   * model, you SHOULD write/generate your own key function, tailored to your
-   * language.
-   */
-  function defaultWordform2Key(wordform: string): SearchKey {
-    // Use this pattern to remove common diacritical marks.
-    // See: https://www.compart.com/en/unicode/block/U+0300
-    const COMBINING_DIACRITICAL_MARKS = /[\u0300-\u036f]/g;
-    return wordform
-      .normalize('NFD')
-      .toLowerCase()
-      // remove diacritical marks.
-      .replace(COMBINING_DIACRITICAL_MARKS, '') as SearchKey;
-  }
+/**
+ * Converts wordforms into an indexable form. It does this by
+ * normalizing the letter case of Latin characters and removing
+ * common diacritical marks.
+ *
+ * This is a very naïve implementation, that I only think will work on
+ * some language that use the Latin script. As of 2020-04-08, only
+ * 4 out of 11 (36%) of published language models use the Latin script,
+ * so this might not actually be a great default.
+ *
+ * This uses String.prototype.normalize() to convert normalize into NFD.
+ * NFD is an easy way to separate a Latin character from its diacritics;
+ * Even then, some Latin-based orthographies use code points that,
+ * under NFD normalization, do NOT decompose into an ASCII letter and a
+ * combining diacritical mark (e.g., SENĆOŦEN).
+ *
+ * Use this only in early iterations of the model. For a production lexical
+ * model, you SHOULD write/generate your own key function, tailored to your
+ * language.
+ */
+export function defaultWordform2Key(wordform: string): string {
```

Comment:
```suggestion
export function defaultWordFormToKey(wordform: string): string {
```

Is there any way we can avoid having this function in two places? It seems it should belong in its own file in /common/web/?

Reply:
I see that this function uses `normalize` but the one in trie-model.ts still uses PARTIAL_NFD_LOOKUP.

Reply:
Please see #3014.

Reply:
>Is there any way we can avoid having this function in two places? It seems it should belong in its own file in /common/web/?

Not... really. Already compiled models in the lexical-models repo _depend_ on a `searchTermToKey` function being bundled with Keyman. Additionally, this differs _slightly_ in the approach of the default provided in the compiler.

Reply:
It _would_ be easier to share code if we got the Lerna stuff figured out. Then the default search term to key function could be a micro module that is depended upon by both the compiler and the LMLayer. It can be _bundled_ in the LMLayer.

Reply:
I'm already seeing some merit to establishing a few "micro-modules" in other areas too.

Reply:
:+1: for micro modules. As long as we keep the dependency tree sane!